CQLCookbook.md

```diff
@@ -0,0 +1,349 @@
+# CQL Cookbook
+Reference for creating a CQL prepopulation file for a DRLS Ruleset.
+
+
+## Overview
+As described in the Ruleset Development 101 document, DRLS invovles the prepopulation of clinical questionnaires from a patient's electronic health record (EHR). This prepopulation is done by means of a clinical quality language (CQL) file that fetches FHIR resources form the EHR.
+
+## Contents
+Every ruleset within the CDS-Library repo will contain at least one CQL prepopulation file. These CQL files tend to follow the same generic format. The instructions below will provide:
+1. [CQL Template](#1-cql-template): A generic template to set up your DRLS prepopulation file.
+   - A) Header
+   - B) Codesystems
+   - C) Value Sets
+   - D) Codes
+   - E) Request Parameter
+   - F) Patient Context
+   - G) Define Statement
+      - "Query" Statements
+      - "Function" Statements
+2. [DRLS Statement Templates](#2-drls-statement-templates): A list of frequently used functions that can be copied and modified to be used within your respective prepopulation file. 
+   - I) [Condition Resource Statements](#i-condition-resource-statements)
+      - *List of All Active Conditions*
+      - *List of Patient's 'Other Diagnoses'*
+      - *List of All Relevant Conditions (as specified by a partiular value set)*
+   - II) [Observation Resource Statements](#ii-observation-resource-statements)
+      - *Extract a Numeric Value of an Observation*
+      - *Extract the date of an Observation*
+      - *Highest Numerical Lab Result*
+      - *Extract performer field of Observation (the performer field has a value of an array with references)*
+3. [Example Prepopulation.cql file](#3-example-prepopulationcql-file) An example of a prepopulation DRLS file.
+4. [Links and Other Resources](#4-links-and-other-resources)
+
+***
+
+## 1) CQL Template
+The structure of your CQL prepopulation file will generally follow the format below:
+
+*Notes:*
+  - Most of the information from this section is described in futher detail within HL7's [CQL Authoring Guide](https://cql.hl7.org/02-authorsguide.html).
+  - The code excerpts below are taken from the [Home Oxygen Therapy Prepopulation file](https://github.com/HL7-DaVinci/CDS-Library/blob/master/HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql).
+
+### A) Header
+Name library, define FHIR version, and include any helper libraries.
+```sql
```

Comment:
Why are we referring to sql here?

Reply:
So technically "cql" syntax highlighting is supported by Github markdown, but it is just an [extension of sql](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml) (so it ends up displaying the exact same way). Also I don't believe 'cql' issupported in non-Github markdown (it does not work on my VSCode editor), so I have been using 'sql' so that people can still see the syntax highlighting if opening the README on a non-Github platform.